megamek/src/megamek/client/ui/swing/unitDisplay/WeaponPanel.java

```diff
@@ -1195,12 +1195,10 @@ public void displayMek(Entity en) {
                     && game.getPhase().isFiring()) {
                 hasFiredWeapons = true;
                 // add heat from weapons fire to heat tracker
-                if (entity.usesWeaponBays()) {
+                if (entity.isLargeCraft()) {
```

Comment:
Are we _absolutely certain_ that this only applies to Large Craft, not other Aerospace units that can use bays?
Apparently Small Craft, Fighter Squadrons, and individual ASFs have all had the option to use bays at one point or another, but I'm not well-enough versed to say if that applies here or not.

Reply:
Fighter Squadrons in particular use bays (at least tabletop-rules-wise). However, a recent-ish ruling explicitly says that ASF and SC should _not_ use weapon bays otherwise: https://bg.battletech.com/forums/index.php?topic=72939

Reply:
> Are we _absolutely certain_ that this only applies to Large Craft, not other Aerospace units that can use bays? Apparently Small Craft, Fighter Squadrons, and individual ASFs have all had the option to use bays at one point or another, but I'm not well-enough versed to say if that applies here or not.

Good catch! I had made a helper method already that'll check if a weapon is a bay, and then either add the heat for the weapon if it isn't a bay, or add the heat for all the bay weapons if it is a bay. 

This if statement's "else" has been modified to handle it - if some entity, for some reason, has weapon bays but isn't a large craft, it should be able to properly add its heat now.

I'm going to check my other changes to make sure this same logic applies there.

Reply:
I updated this location & another to use the getHeatByBay to ensure it still works. The other locations where I changed checking if a has a weapon bay to checking if it's a large craft are truly only for large craft.